### PR TITLE
Minor fixes for ServiceAccountIssuerDiscovery

### PR DESCRIPTION
**Pull request by mtaufen**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

- Drop the test tag so the tests don't just run in the alpha job.
- Add some indication of API stability to the OpenAPI docs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/routes/openidmetadata.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (s *OpenIDMetadataServer) Install(c *restful.Container) {
 	cfg.Path(serviceaccount.OpenIDConfigPath).Route(
 		cfg.GET("").
 			To(fromStandard(s.serveConfiguration)).
-			Doc("get service account issuer OpenID configuration, also known as the 'OIDC discovery doc'").
+			Doc("get service account issuer OpenID configuration, also known as the 'OIDC discovery doc'. Beta.").
```
Comment (liggitt) [inline pkg/routes/openidmetadata.go:70]: Suggest something like "This is a beta endpoint that requires the feature gate ServiceAccountIssuerDiscovery to be enabled."

**Comment by liggitt** [comment]
something like this would allow routes to decorate themselves with alpha/beta/stable tags:

```diff
diff --git a/api/openapi-spec/swagger.json b/api/openapi-spec/swagger.json
index a618dc27427..ecc417d373e 100644
--- a/api/openapi-spec/swagger.json
+++ b/api/openapi-spec/swagger.json
@@ -18927,7 +18927,8 @@
           "https"
         ],
         "tags": [
-          "WellKnown"
+          "WellKnown",
+          "beta"
         ]
       }
     },
@@ -105152,7 +105153,8 @@
           "https"
         ],
         "tags": [
-          "openid"
+          "openid",
+          "beta"
         ]
       }
     },
diff --git a/pkg/routes/openidmetadata.go b/pkg/routes/openidmetadata.go
index 9ee8b1f175e..007b43d608c 100644
--- a/pkg/routes/openidmetadata.go
+++ b/pkg/routes/openidmetadata.go
@@ -67,6 +67,7 @@ func (s *OpenIDMetadataServer) Install(c *restful.Container) {
 	cfg.Path(serviceaccount.OpenIDConfigPath).Route(
 		cfg.GET("").
 			To(fromStandard(s.serveConfiguration)).
+			Metadata("stability", "beta").
 			Doc("get service account issuer OpenID configuration, also known as the 'OIDC discovery doc'").
 			Operation("getServiceAccountIssuerOpenIDConfiguration").
 			// Just include the OK, doesn't look like we include Internal Error in our openapi-spec.
@@ -80,6 +81,7 @@ func (s *OpenIDMetadataServer) Install(c *restful.Container) {
 	jwks.Path(serviceaccount.JWKSPath).Route(
 		jwks.GET("").
 			To(fromStandard(s.serveKeys)).
+			Metadata("stability", "beta").
 			Doc("get service account issuer OpenID JSON Web Key Set (contains public token verification keys)").
 			Operation("getServiceAccountIssuerOpenIDKeyset").
 			// Just include the OK, doesn't look like we include Internal Error in our openapi-spec.
diff --git a/staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go b/staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go
index e3bd028bbf9..c7753db9445 100644
--- a/staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go
+++ b/staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go
@@ -85,6 +85,20 @@ func GetOperationIDAndTags(r *restful.Route) (string, []string, error) {
 	} else if len(parts) >= 1 {
 		tags = append(tags, ToValidOperationID(parts[0], false))
 	}
+
+	if stability, ok := r.Metadata["stability"]; ok {
+		switch stability {
+		case "alpha":
+			tags = append(tags, "alpha")
+		case "beta":
+			tags = append(tags, "beta")
+		case "stable":
+			tags = append(tags, "stable")
+		default:
+			return "", nil, fmt.Errorf("unknown stability level: %v", stability)
+		}
+	}
+
 	return prefix + ToValidOperationID(op, prefix != ""), tags, nil
 }
 
```

File: pkg/routes/openidmetadata.go (hunk 1 above)
Comment (mtaufen) [reply pkg/routes/openidmetadata.go]: Done.

**Comment by mtaufen** [comment]
Updated the doc text.

I dig the Metadata approach. @hh WDYT?
Any other stakeholders we should get consensus with before we do that?

**Comment by hh** [comment]
I am keen, we already did a bit of research to prepare for this.

**Comment by hh** [comment]
I wonder if we could update the metadata for all apis, so we have one place to clarify the designated status of an API?

**Comment by liggitt** [comment]
needs rebase, might be obsolete now

**Comment by desponda** [comment]
Hello! I am from the Bug Triage team! Friendly reminder that code freeze is starting March 9th, 2021 (about 1 week from now). We want to ensure that each PR has a chance to be merged before code freeze.
As the PR is tagged for 1.21, is it still planned for this release?

**Comment by mtaufen** [comment]
The actual code in this PR is obsolete, the test already got promoted to conformance and had metadata added, and the docstring changes this PR proposes aren't accurate now that the feature is GA. I'm going to close it. @hh if we want to add tags as @liggitt proposed, let's open a separate issue.

**Pull request closed without merging**